### PR TITLE
Fix issue when run command with huge stdout data

### PR DESCRIPTION
**Pull request by ericclappier-eaton**

### CODE REVIEW

**Review by aquette — APPROVED** [review]

File: lib/src/fty_common_nut_utils_private.cc [hunk 1]
```diff
@@ -28,44 +28,69 @@ int runCommand(
...
+        logError("{}", pid.error());
         return -1;
     }
 
-    auto ret = proc.wait(msTimeout);
-    if (!ret) {
-        log_error(ret.error().c_str());
-        return -1;
+    const int WAIT_TIMEOUT_LOOP = 2000;
+    int count = msTimeout / WAIT_TIMEOUT_LOOP;
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/fty_common_nut_utils_private.cc:48]: shall be: int count = ((msTimeout / WAIT_TIMEOUT_LOOP) + 1);
with the if(count)...

File: lib/src/fty_common_nut_utils_private.cc [hunk 2]
```diff
@@ -28,44 +28,69 @@ int runCommand(
...
+    if (count < 1) count = 1;
+    bool resulOk = false;
+    std::string tmp;
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/fty_common_nut_utils_private.cc:51]: shall be named tmpOutput instead

File: lib/src/fty_common_nut_utils_private.cc [hunk 3]
```diff
@@ -28,44 +28,69 @@ int runCommand(
...
+    while (count > 0) {
+        auto res = proc.wait(WAIT_TIMEOUT_LOOP);
+        if (!res && res.error() == "timeout") {
+            count --;
+            tmp += proc.readAllStandardOutput();
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/fty_common_nut_utils_private.cc:56]: accumulate stdout in a tmp...
why not do the same thing with stderr? (it is used later)

File: lib/src/fty_common_nut_utils_private.cc [hunk 4]
```diff
@@ -28,44 +28,69 @@ int runCommand(
...
+            logTrace("Wait process timeout, count={} tmp.size={}", count, tmp.size());
+            continue;
+        } else if (!res) {
+            logWarn("Cannot wait process, error: {}", res.error());
+            break;
+        } else {
+            tmp += proc.readAllStandardOutput();
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/fty_common_nut_utils_private.cc:63]: here, accumulate stderr too...

File: lib/src/fty_common_nut_utils_private.cc [hunk 5]
```diff
@@ -28,44 +28,69 @@ int runCommand(
...
+            resulOk = (*res == 0);
+            break;
+        }
     }
 
-    stdout = proc.readAllStandardOutput();
+    if (resulOk) {
+        stdout = tmp;
+    }
     stderr = proc.readAllStandardError();
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/fty_common_nut_utils_private.cc:73]: here, handle accumulation of stderr done in loop (without 'proc' usage)

File: lib/src/fty_common_nut_utils_private.cc [hunk 6]
```diff
@@ -28,44 +28,69 @@ int runCommand(
...
-        log_error(ret.error().c_str());
-        return -1;
+    const int WAIT_TIMEOUT_LOOP = 2000;
+    int count = msTimeout / WAIT_TIMEOUT_LOOP;
+    if (count < 1) count = 1;
+    bool resulOk = false;
+    std::string tmp;
+    while (count > 0) {
+        auto res = proc.wait(WAIT_TIMEOUT_LOOP);
+        if (!res && res.error() == "timeout") {
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/fty_common_nut_utils_private.cc:54]: the timeout error case shall be handled in a 'if (!res)' block, as a continued case VS break...

File: lib/src/fty_common_nut_utils_private.cc (hunk 2 above)
Comment (ericclappier-eaton) [reply lib/src/fty_common_nut_utils_private.cc:51]: Take into account

File: lib/src/fty_common_nut_utils_private.cc (hunk 3 above)
Comment (ericclappier-eaton) [reply lib/src/fty_common_nut_utils_private.cc:56]: Take into account

File: lib/src/fty_common_nut_utils_private.cc (hunk 4 above)
Comment (ericclappier-eaton) [reply lib/src/fty_common_nut_utils_private.cc:63]: Take into account

File: lib/src/fty_common_nut_utils_private.cc (hunk 5 above)
Comment (ericclappier-eaton) [reply lib/src/fty_common_nut_utils_private.cc:73]: Take into account

File: lib/src/fty_common_nut_utils_private.cc (hunk 6 above)
Comment (ericclappier-eaton) [reply lib/src/fty_common_nut_utils_private.cc:54]: Take into account

**Pull request merged**